site/docs/blog/posts/2024-05-cache-enhancements.md

```diff
@@ -0,0 +1,62 @@
+stores the current "tip" of branches and tags, and the `objs` ("objects") table, which stores all commits and content
+objects. Objects in Nessie are (usually, see below) immutable. Immutable objects are great candidates to be cached - so
+we did exactly that: cache the contents of the `objs` table. With "just" Caching for the `objs` table we see very few
+reads against that table in production systems.
+
+However, nothing is cached for the references (`refs`) table. This means that basically all Nessie API calls hit the
+database, no matter whether it is "just" a read operation like "get-the-tip-of-a-branch" or a commit operation. Given
+that most operations against a lakehouse are reads, we could save a lot of read operations if there's a way to cache
+references.
+
+The [PR to cache references](https://github.com/projectnessie/nessie/pull/8111) introduced the capability to cache
+references, but only in the "local" Nessie instance. A commit is immediately visible as a change to the "tip"/HEAD
+of the branch on the Nessie instance that performed the commit - but another Nessie instance serving the same Nessie
+repository would not know that the reference has changed and "happily" serve and work with the stale cache information.
+This was a known (and for development/"cognitive review burden" reasons accepted) limitation for that PR, and the most
+important reason why the functionality to cache references is labeled "experimental" and is disabled by default. It
+would be fine to enable it, but only if **only** a single Nessie instance accesses the repository.
+
+[Distributed cache invalidation](https://github.com/projectnessie/nessie/pull/8463) introduced the ability for Nessie
+instances to tell other Nessie instances to invalidate certain cached objects, so that the cache stays consistent. A
+cache invalidation message instructs a Nessie instance to remove certain key(s) from its cache.
+
+Cache invalidation messages are rather "management" concerns and the endpoints, even if protected by any mechanism,
+should not be publicly exposed. Thankfully Quarkus recently added the ability to [separate management endpoints from
+public ones](https://quarkus.io/guides/management-interface-reference), Nessie incorporated this in the 0.82.0 release -
+management endpoints are available on port 9000 (default) and the Nessie REST APIs stay on the known port 19120. In
+other words, management endpoints are not publicly available, which is what we want.
+
+The Nessie endpoint to receive cache invalidation messages is only available on the management port. If you deploy
+Nessie via the [Helm chart](../../guides/kubernetes.md), or via our
+[Kubernetes operator](https://github.com/projectnessie/nessie/pull/7967) (not released yet). If you deploy Nessie via
+our Helm chart, there's nothing you need to configure for distributed cache invalidations - it's setup and configured
+automatically.
```

Comment:
nit: I tend to think it might be worth specifically mentioning that the eviction service relies on discovering an IP per Nessie node. K8s deployments differ, this notice could save some pain for administrators if they know it up front.

Reply:
Updated this and docs